scipy/optimize/_linprog.py

```diff
@@ -207,13 +207,17 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
         The coefficients of the linear objective function to be minimized.
     A_ub : 2-D array, optional
         The inequality constraint matrix. Each row of ``A_ub`` specifies the
-        coefficients of a linear inequality constraint on ``x``.
+        coefficients of a linear inequality constraint on ``x``. Accepts sparse 
+        matrix. If method="simplex" or "revised simplex", tries to convert
+        sparse matrix to dense.
```

Comment:
We can remove "Accepts sparse matrix" in favor of putting "...or sparse matrix" above, where the type is stated. 

Instead of 
> If method="simplex" or "revised simplex", tries to convert sparse matrix to dense.

let's put "Converts sparse input to dense array." in the equivalent spot of the `simplex` and `revised simplex` 
documentation only.

So no changes in this part of the top-level documentation. Please only a note about converting to dense in the `simplex` and `revised simplex` method-specific documentation.